### ISSUE TITLE
Proposal for Code of Conduct 

### ISSUE THREAD

**Issue by iceLearn** [crowdresearch/collective#11]
**---- Edited proposal after Consensus [meeting ](https://www.youtube.com/watch?v=UmGGKwuMp0g&feature=youtu.be)--------------**

Without a formal code of conduct, our norms are challenging to enforce. A majority of online communities have some kind of code of conduct however the Crowd Research Collective does not have one that targets the specific needs we face.

We, as a collective have been working on a drafted CoC in this document - https://docs.google.com/document/d/1hpBa460MFoG5qsKatxVPmdpF77VITetA-YGp7V3KlRY/edit#

Since we need to ratify the draft and adopt the CoC to our governance model, on behalf of the SCRC, I would like to formally introduce a collective proposal - 

# Proposal 
1. We will attempt series of meetings to discuss the content in the draft. Meetings will be announced in the SLACK #announcement channel.  
2. While we are having meetings, will introduce some deadlines to comment/ express concerns  in the document so it will be easier to ratify and will be inclusive since not all the members will be able to make to meetings. 
3. Once we all agree upon the content, the CoC document will be added to the SCRC repository where every member who join the group will be covered by it.

**Use comments to share your response** or use emoji 👍 to show your support. **To officially join in**, add yourself as an assignee to the proposal. **To break consensus**, comment using [this template](https://raw.githubusercontent.com/crowdresearch/collective/master/docs/breaking_consensus). To find out more about this process, read the [how-to](https://github.com/crowdresearch/collective/blob/master/docs/CONTRIBUTING.md).

---

**-----This is the Initial proposal which has no consensus  -------------------------------------**
Without a formal code of conduct, our norms are challenging to enforce. A majority of online communities have some kind of code of conduct however the Crowd Research Collective does not have one that targets the specific needs we face.

### My proposal
I propose a code of conduct based on the draft process in this [document](https://docs.google.com/document/d/1hpBa460MFoG5qsKatxVPmdpF77VITetA-YGp7V3KlRY/edit#) to be singed by every existing member of the collective and every new member to our community. The draft process is ongoing, so please participate by adding comments here, or by adding comments or suggestions to the linked document.
Our ultimate goal would be to host the ratified Code of Conduct in this repo -https://github.com/crowdresearch/collective/blob/master/docs/CODE_OF_CONDUCT.md

---
**Use comments to share your response** or use emoji 👍 to show your support. **To officially join in**, add yourself as an assignee to the proposal. **To break consensus**, comment using [this template](https://raw.githubusercontent.com/crowdresearch/collective/master/docs/breaking_consensus). To find out more about this process, read the [how-to](https://github.com/crowdresearch/collective/blob/master/docs/CONTRIBUTING.md).

**Comment by markwhiting** [comment]
I've just been reviewing the [Contributor Covenant](https://www.contributor-covenant.org) as a starting point, and added it in to begin draft 5. I'm going back through to edit around specific needs that we have previously expressed beyond what this solutions includes.

**Comment by markwhiting** [comment]
I did another pass this morning to try to ensure things were aligned around some of the core ideas we had previously discussed.

I will do more passes but please jump in and add comments and suggestions.

**Comment by mbernst** [comment]
I left a few comments! I know that Angela had previously provided some really useful references to other documents. I wonder if we might at least include some related reading to keep that linked somewhere in the doc.

Is it possible to create an addendum to a proposal? For example, I would propose that this document, once ratified, be included in our gov docs in this repo.

**Comment by markwhiting** [comment]
Yep. that was the intention. Perhaps @iceLearn you could edit to say that the intention would be to host it here: https://github.com/crowdresearch/collective/blob/master/docs/CODE_OF_CONDUCT.md

**Comment by iceLearn** [comment]
Yes, added that.. will do a pass today.

**Comment by neilthemathguy** [comment]
**BREAK THE CONSENSUS** 

The code of conduct is in-progress document written by many people in the project. Similar to other governance documents, it is a joint effort on which several members of the collective have worked for many hours. There are planned hangouts to discuss, debate, and vote on the code of conduct. This is not new work and should not be submitted as a proposal. It is important to note that no single individual can claim the authorship of this or other governance documents.  Once we collectively draft and agree on the code of conduct, we can add it to the common repository under https://github.com/crowdresearch/daemo

**Comment by iceLearn** [comment]
> The code of conduct is in-progress document written by many people in the project. Similar to other governance documents, it is a joint effort on which several members of the collective have worked for many hours. There are planned hangouts to discuss, debate, and vote on the code of conduct. 

Yes, totally agree on this and there is no doubt about it. We all drafted this, commented, included things which we thought for the betterment of the crowd. You are still able to make any changes to the document or request for more clarifications and also take part in the hangouts in the collective on CoC.

> This is not new work and should not be submitted as a proposal

Disagreeing to this point because our governs model we never said only new proposals should be submitted. Besides we need to have CoC to our collective and this proposal is to include/ introduce and request for more collaboration to the ongoing draft. 
By new work did you mean new document? new ideas? if so, we have Draft 5 now based in the draft 4 comments and clarifications 
at the same time I would like to ask if you have a better plan how might we include a CoC to this collective other than proposing here? 

> It is important to note that no single individual can claim the authorship of this or other governance documents.
There is no single individual calming authorship for this document rather requesting individuals to collaborate in the process.   Once we collectively draft and agree on the code of conduct, we can add it to the common repository under  https://github.com/crowdresearch/daemo

Yes, "Agreed CoC" should be added to a common repository, I dont have disagreement to the repository link, I guess we are in the process of logistical arrangements and if thats the link, lets upload it there!

**Comment by neilthemathguy** [comment]
Thanks for checking! Let me elaborate. There are two parts to this. First, do we need CoC?: my answer is YES. Second, does CoC needs a proposal by an individual: No.    

> _By new work did you mean new document? new ideas? if so, we have Draft 5 now based in the draft 4 comments and clarifications_

By “not new work”, I mean it is already in progress, we have agreed on its importance, and it is ongoing. We were writing CoC in parallel with the Governance documents; we just prioritized and pushed the Governance documents to GitHub first, CoC is listed in it. [LINK](https://github.com/crowdresearch/collective/blob/master/governance/process.md#code-of-conduct) 

As the Governance Documents didn't need proposal by an individual, CoC doesn't need it either.  

> _at the same time I would like to ask if you have a better plan how might we include a CoC to this collective other than proposing here?_

This will follow the same process as previous Governance documents. That was the reason we organized the hangouts. The goal was to discuss, debate, finalize the version of CoC, open it for voting, and then include it directly in enforced governance documents.  

Also, at present, the way above proposal is written, it looks like one individual has proposed the need  of CoC for the project. The proposal #11, starts as _“I propose a code of conduct based on the draft process in this”_ , which as we all know is not the case. 

I agree with @mbernst as he mentioned above:
>  I would propose that this document, once ratified, be included in our gov docs in this repo.

So let’s do the hangouts, and complete CoC. Once we finish the voting, we can add CoC to the governance docs directly.

**Comment by shirishgoyal** [comment]
> if so, we have Draft 5 now based in the draft 4 comments and clarifications

It is important to discuss all of this on hangout. These versions are not iterations from the previous work but appear to be significantly different  from what was done in the document as a community and we didn’t have any hangout discussions regarding this yet. Let’s discuss further in hangout.

**Comment by markwhiting** [comment]
I think the reason a proposal seems right here is that we don't have a formal way to modify documents other than the proposal mechanism.

@neilthemathguy I agree with your point about the way proposals are written makes them look like they are the voice of an individual. Perhaps we could adjust the instructions so that we don't have this problem in the future. e.g. we might suggest the framing 'One solution is...' or something like that.

**Comment by neilthemathguy** [comment]
#21 already covers people's contributions they can point to it even if they want to stay inactive in the project. 

I would like to know why do you think discharging volunteer contributors is important for the Collective?

**Comment by neilthemathguy** [comment]
I think the issue is unintentional misattribution and it is bigger than framing the proposal as "one solution is ...".

We should clearly state how this proposal came in a picture and who should post it--- Collective has been working on this for months, and has planned hangouts to discuss it further. This is ongoing work, not finished yet, nor approved by the Collective, yet.

In the future, before posting any proposals of the collective work, it should be communicated upfront to the Collective. The proposal should clearly mention that its been added on the behalf of the Collective. In any case, Collective effort should not appear as a proposer’s solo work.

**Comment by markwhiting** [comment]
I agree with the misattribution point. That said, I think given our governance mechanism, we should use the proposal mechanism to formalize this document.

**Comment by markwhiting** [comment]
@neilthemathguy I think you intended to break consensus here. Could you edit to use the template for that which is linked in the original post and also [here](https://raw.githubusercontent.com/crowdresearch/collective/master/docs/breaking_consensus).

**Comment by mbernst** [comment]
Just reiterating that there is the question about how this effort is described in the proposal, and about how it gets ratified. I'm focusing on the latter here.

I agree that our governance approach indicates adopting a change at this point requires consensus on a proposal, or a vote if consensus is broken. Thinking on this, consensus on the document is stronger (numerically speaking) than a vote of the collective. So my assumption is that we iterate until we have consensus, or until we deadlock.

Go team!

**Comment by iceLearn** [comment]
In order to bring everyone to the same thought should we - 
1. re-frame  this proposal as a collective not to mean as a individual but we all refine it, the proposal is to add the coc to the governance structure - things which need to add, we discuss in consensus meeting 
2. Re-submit the CoC once we all have CoC which everyone agreed , based on the hangouts and comments 
What is your idea?

**Comment by markwhiting** [comment]
Its seems not possible to get everyone in one meeting, even if we schedule it so that everyone who responds can attend, we could still have someone who later says their opinion wasn't heard. For this reason, it seems like a better process regarding your (@iceLearn) point 2 would be to do something like have a collectively authored file that people can edit up until a deadline, then go through a process of trying to resolve any remaining conflicts.

**Comment by iceLearn** [comment]
# Summary

Based on the consensus meeting 20th Dec 2017 https://www.youtube.com/watch?v=UmGGKwuMp0g&feature=youtu.be (1:28:20)  we all agreed on 
1- We need the proposal in order to adopt the ratified Code of Conduct under new governance structure
2- Already submitted proposal need to re-frame to reflect the CoC is a collective effort 
3- Proposal should be edited and emphasize that there will be a collective process to ratify the drafted CoC document via series of meeting and having deadlines to comment on the CoC document 

Please see the proposal section for edited version of the proposal.

**Comment by markwhiting** [comment]
The last update on this was: 

> Hangout: https://youtu.be/f4KGfCMSSug
> Attendees - @markwhiting ( first few parts ) @neilthemathguy @shirishgoyal and @dilrukshi
> 
> Outcome - Based on the previous drafts( specially 4 and 5) we created draft 6 - main idea is to include everything which we think as what we want this crowd to practice.
> 
> We all were in agreement that CoC will have a reporting mechanism. Internal and severe cases will be reported to IRB. 
> 
> We agreed that we can set up some deadlines to amend the document and if possible final meeting.
> 
> Anyone interested will edit the draft using "Suggest mode" - https://docs.google.com/document/d/1hpBa460MFoG5qsKatxVPmdpF77VITetA-YGp7V3KlRY/edit#heading=h.1zpxkl4dlhar

Are there any other next steps? What else is holding this back?
